### ISSUE TITLE
Request to query strains with a given plasmid, then retrieve sequences

### ISSUE THREAD

**Issue by mwes** [SD2E/synbiohub_adapter#63]
Per discussion here: https://sd2e.slack.com/archives/C5W4Y826Q/p1549579295335000

@dvnstrcklnd would like to:

"Find strains and plasmids from YS and audit them"

e.g. "I want to find all strains with `YG_plasmid_010` and then get all the plasmids from those strains, then get all their sequences and componens..."

**Comment by nroehner** [comment]
@dvnstrcklnd @mwes The master branch has now been updated with new versions of the query_design_strains method and query_design_plasmids method in query_synbiohub.py. I'm currently trying to figure out how to make these available in TACC's Jupyter Notebook environment, but below is some example to code to try out with a local installation of synbiohub_adapter. DM me on Slack if you do not know the password the SD2 program's SynBioHub account.

```
from query_synbiohub import SynBioHubQuery

sbh_query = SynBioHubQuery('https://hub.sd2e.org')
sbh_query.login('sd2e', '[replace with password for SD2 program's SynBioHub account]')

strains = sbh_query.query_design_strains(pretty=True, with_plasmids=True)

plasmid_sequences = {}
plasmid_features = {}

for key in strains.keys():
    if isinstance(strains[key], list):
        plasmids = strains[key]
    else:
        plasmids = [strains[key]]
    
    plasmid_sequences.update(sbh_query.query_design_plasmids(pretty=True, with_sequence=True, plasmids=plasmids))
    
    plasmid_features.update(sbh_query.query_design_plasmids(pretty=True, with_features=True, plasmids=plasmids))
    
print(len(strains.keys()))
print(strains.keys())

print(len(plasmid_sequences.keys()))
print(plasmid_sequences.keys())

print(len(plasmid_features.keys()))
print(plasmid_features.keys())
```

**Comment by dvnstrcklnd** [comment]
Thanks. This is super helpful. I think identifying key methods like this and putting them into some examples would go quite a long way. I don't have a local installation, so I'll wait until the update gets pulled to TACC.

**Comment by tcmitchell** [comment]
Is this issue complete? Was this work done by #67? Should this issue be closed?

**Comment by mwes** [comment]
(I was just the middleman on this request, I'll leave it to @nroehner and @dvnstrcklnd to confer and close)

**Comment by tcmitchell** [comment]
Let's not close issues without comment. Let's instead communicate to users and future programmers why and how something was addressed, or why it won't be addressed. I'm reopening to allow @nroehner to explain why this was closed. Feel free to close with that explanation.

**Comment by nroehner** [comment]
This issue was addressed by #67. There are examples of using the new query methods in shared Python notebooks on the SD2 Jupyter environment (nroehner-novel_chassis_queries and nroehner-yeast_states_queries).

I will post examples of the code and a link to the README for shared notebooks in the SD2 Jupyter environment once the latter is back up (I'm currently getting a 500 : Internal Server Error when I try to access it).

**Comment by nroehner** [comment]
This issue was addressed by #67. There are examples of using the new query methods in shared Python notebooks on the SD2 Jupyter environment (nroehner-novel_chassis_queries and nroehner-yeast_states_queries). The README for shared notebooks is available at 

https://github.com/SD2E/sd2nb-app/blob/master/README.md

The yeast_states_queries notebook is also available on GitLab at 

https://gitlab.sd2e.org/nroehner/yeast_states_queries
